src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PublicDataRetrievalService.java

```diff
@@ -0,0 +1,76 @@
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.officers.OfficersApi;
+import uk.gov.companieshouse.overseasentitiesapi.client.ApiClientService;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+
+@Component
+public class PublicDataRetrievalService {
+    public static final String COMPANY_NUMBER = "company_number";
+    public static final String OFFICER_NAME = "officer_name";
+
+    private final ApiClientService apiClientService;
+
+    public PublicDataRetrievalService(ApiClientService apiClientService) {
+        this.apiClientService = apiClientService;
+    }
+
+    public void initialisePublicData(String companyNumber, String passThroughTokenHeader) throws ServiceException {
+        getCompanyProfile(companyNumber, passThroughTokenHeader);
+        getOfficers(companyNumber, passThroughTokenHeader);
+    }
+
+    private CompanyProfileApi getCompanyProfile(String companyNumber, String passThroughTokenHeader) throws ServiceException {
+        try {
+            CompanyProfileApi companyProfileData = apiClientService
+                    .getOauthAuthenticatedClient(passThroughTokenHeader).company()
+                    .get("/company/" + companyNumber).execute().getData();
+
+            var logMap = new HashMap<String, Object>();
+            logMap.put(COMPANY_NUMBER, companyProfileData.getCompanyNumber());
+
+            ApiLogger.info("Retrieving Company Profile for Company Number " + logMap);
+
+            return companyProfileData;
+        } catch (URIValidationException | IOException e) {
+            var message = "Error Retrieving Company Profile data for " + companyNumber;
+            ApiLogger.errorContext(message, e);
+            throw new ServiceException(e.getMessage(), e);
+        }
+    }
+
+    private OfficersApi getOfficers(String companyNumber, String passThroughTokenHeader) throws ServiceException {
+        try {
+            ApiResponse<OfficersApi> officersApiResponse = apiClientService
+                    .getOauthAuthenticatedClient(passThroughTokenHeader).officers()
+                    .list("/company/" + companyNumber + "/officers").execute();
+
+            var officersData = officersApiResponse.getData();
+
+            var logMap = new HashMap<String, Object>();
+
+            if (officersApiResponse.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
+                logMap.put(COMPANY_NUMBER, companyNumber);
+                ApiLogger.info("No Officer for Company Number " + logMap);
+            } else {
+                logMap.put(OFFICER_NAME, officersData.getItems().get(0).getName());
```

Comment:
It's not one office name. It's a list of officers being returned. You don't need to log each officer (except for test evidence)

Reply:
Maybe a loop and debug logs for each officer name - but not info. As we should never log personal information.

Reply:
I couldnt find anything in https://github.com/companieshouse/structured-logging/blob/master/src/main/java/uk/gov/companieshouse/logging/util/DataMap.java to use as a key for a list of officers. Any suggestions for that? @AndreKramer7654

Reply:
I'd just log the count of officers returned. With the company number in map and the count just in the log message.